crates/net/discv5/src/metrics.rs

```diff
@@ -0,0 +1,137 @@
+//! Tracks peer discovery for [`Discv5`](crate::Discv5) and
+//! [`Discv5BCv4`](crate::Discv5BCv4`).
+
+use metrics::Counter;
+use reth_metrics::Metrics;
+
+use crate::{ChainRef, IdentifyForkIdKVPair};
+
+#[doc(hidden)]
+pub trait UpdateMetrics {
+    fn with_metrics<R, F>(&mut self, f: F) -> R
+    where
+        F: Fn(&mut Metrics) -> R;
+}
```

Comment:
why do we need a trait for this? we should be able to include metrics structs inside of components